# HG changeset patch
# User Rik <rdrider0-list@yahoo.com>
# Date 1260479365 28800
#      Thu Dec 10 13:09:25 2009 -0800
# Node ID 9bd50882cd1c21d861772342b777e500c1772acc
# Parent  dbbec33d14f8cea25dad333afeda7ca7c44c0bd1
new distdir-local rule stops package distribution unless libraries necessary to create documentation are present.

diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,8 +1,13 @@
+2009-12-10  Rik <octave@nomad.inbox5.com>
+
+	* interpreter/Makefile.am: new distdir-local rule to check for presence of
+	necessary libraries before packaging distribution
+
 2009-12-08  Rik <octave@nomad.inbox5.com>
 
 	* interpreter/Makefile.am: Distribute all generated documentation (.pdf,
 	.ps, .html, .dvi) to remove build requirement for TexInfo
 
 2009-12-07  Rik <octave@nomad.inbox5.com>
 
 	* interpreter/Makefile.am, faq/Makefile.am, liboctave/Makefile.am: Document
diff --git a/doc/interpreter/Makefile.am b/doc/interpreter/Makefile.am
--- a/doc/interpreter/Makefile.am
+++ b/doc/interpreter/Makefile.am
@@ -151,16 +151,25 @@ octave.info: $(IMAGES_TXT)
 octave.dvi octave.ps: $(IMAGES_EPS)
 
 octave.html: $(IMAGES_PNG)
 
 octave.pdf: $(IMAGES_PDF)
 
 all-local: dvi html pdf ps doc-cache $(HTMLDIR_IMAGES)
 
+# Prevent packaging of distribution unless all libraries 
+# necessary to create documentation are present
+distdir: distdir-local
+distdir-local:
+	@$(GREP) '#define HAVE_COLAMD 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing COLAMD library.  Cannot package distribution!" ; exit 1; }
+	@$(GREP) '#define HAVE_CHOLMOD 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing CHOLMOD library.  Cannot package distribution!" ; exit 1; }
+	@$(GREP) '#define HAVE_UMFPACK 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing UMFPACK library.  Cannot package distribution!" ; exit 1; }
+	@$(GREP) '#define HAVE_QHULL 1' $(top_builddir)/config.h > /dev/null || { echo "Documentation creation requires missing QHULL library.  Cannot package distribution!" ; exit 1; }
+
 # Install doc-cache of help files
 install-data-local:
 	$(MKDIR_P) $(DESTDIR)$(octetcdir)
 	$(INSTALL_DATA) doc-cache $(DESTDIR)$(octetcdir)/doc-cache
 
 uninstall-local:
 	rm -f $(DESTDIR)$(octetcdir)/doc-cache
 
